solver/src/main/java/org/chocosolver/solver/variables/view/graph/directed/DirectedEdgeInducedSubgraphView.java

```diff
@@ -103,7 +103,7 @@ protected boolean doRemoveNode(int node) throws ContradictionException {
         if (enforceNodes.contains(node)) {
             contradiction(this, "Try to remove mandatory node");
         }
-        return false;
+        return true;
```

Comment:
Hmmm, returning `true` suppose that a modification occurs, which is not the case here, right?

Reply:
You are right, it is a non-intuitive shortcut. Actually, to remove a node from an edge-induced subgraph view, it is only necessary to isolate the node by removing every of its incident edge, so all the work is done in the `removeNode` method of the `UndirectedGraphView` (resp. `DirectedGraphView`). As you can see in the following code, the `doRemoveNode` method will only be called if there is no mandatory incident edge (`getMandatoryNodes().contains(node)`) and if there is at least one potential incident edge (`getPotentialNodes().contains(node)`). Thus, when `doRemoveNode(node)` occurs, all incident edges should have already been removed, and the node is consequently also removed, so `doRemoveNode` will always return true. I pushed a new commit with a semantically correct return statement: `return !getPotentialNodes().contains(node);`.

```
    @Override
    public boolean removeNode(int node, ICause cause) throws ContradictionException {
        assert cause != null;
        assert (node >= 0 && node < getNbMaxNodes());
        if (getMandatoryNodes().contains(node)) {
            this.contradiction(cause, "remove mandatory node");
            return false;
        } else if (!getPotentialNodes().contains(node)) {
            return false;
        }
        ISet nei = getPotentialNeighborsOf(node);
        for (int i : nei) {
            removeEdge(node, i, cause);
        }
        if (doRemoveNode(node)) {
            if (reactOnModification) {
                delta.add(node, GraphDelta.NODE_REMOVED, cause);
            }
            GraphEventType e = GraphEventType.REMOVE_NODE;
            notifyPropagators(e, cause);
            return true;
        }
        return false;
    }
```